### DISCUSSION TITLE
[V5] How can I make validations dynamic? And there is problem with preload!

### DISCUSSION THREAD

**Discussion by remidosol** [adonisjs/core#1644] [Help] [answered]
Hello folks.

First problem:
```ts
const advertisementId = params.id
const advertisement = await Advertisement.findBy('id', advertisementId)
```

I use these lines for fetching ONE Advertisement. There is no problem at **Advertisement's @afterFind hook** like following.

```ts
@belongsTo(() => Category)
  public category: BelongsTo<typeof Category>;

@afterFind()
  public static async preloadAllForSingle (advertisement: Advertisement) {
    try {
      await advertisement.preload((preloader)=>{
        preloader
          .preload('user')
          .preload('category') // problem is in this line
      })
    } catch (err) {
      console.log(err.message)
      console.log(err.stack)
    }
  }
```

But when I try to preload category in Advertisement hook, it runs **Category's @afterFetch hook**. _Should not it run **Category's @afterFind hook**?_ This was the first problem.

Second problem is "How can I make validators dynamic?" There are a lot of request body keys that are should validated. Should I define a schema for every key? There are about **180 categories**. How can I validate all? Thanks already.

**Comment by thetutlage** [comment, marked as answer]
Please make dedicated threads for individual issues. Every thread created should help someone in the future :)

**Reply by remidosol** [reply]
Alright then, thanks
